File: api-go/middleware/middleware.go
package middleware

import (
	"api-go/lib"
	"strconv"

	"github.com/gofiber/fiber/v2"
)

func AuthMiddleware(c *fiber.Ctx) error {
	token := c.Get("x-token")
	if token == "" {
		return c.Status(400).JSON(fiber.Map{
			"error": "Missing x-token",
		})
	}

	user, err := lib.VerifyJwt(token)
	if err != nil {
		return c.Status(400).JSON(fiber.Map{
			"error": "Invalid token: " + err.Error(),
		})
	}

	c.Locals("user", user)

	return c.Next()
}

func ParamId(c *fiber.Ctx) error {
	id, err := strconv.ParseInt(c.Params("id"), 10, 64)

	if err != nil {
		return c.Status(400).JSON(fiber.Map{
			"error": "Invalid music ID",
		})
	}

	c.Locals("id", id)

	return c.Next()
}

File: api-node/src/middleware.ts
import { createMiddleware } from "hono/factory"
import { verifyJwt, type Jwt } from "./lib/jwt.js"

export const authMiddleware = createMiddleware<{
    Variables: {
        user: Jwt
    }
}>(async (c, next) => {
    const token = c.req.header("x-token")
    let res
    try {
        res = await verifyJwt(token!)
    } catch (error) {
        return c.json({ error: 'jwt not valid' }, 400)
    }
    c.set('user', res)

    await next()
})

